Yank/pipeline.py

```diff
@@ -1381,6 +1400,17 @@ def _setup_system(self, system_file_path, pack, alchemical_charge,
         self._setup_molecules(*molecule_ids)
         solvent = self.solvents[solvent_id]
 
+        # Start error tracking variables
+        # Water
+        known_solvent_files = [file for file in _OPENMM_LEAP_SOLVENT_FILES_MAP_R.keys()]
+        loaded_water_files = []  # Detected loaded water files
+
+        def extend_list_of_waters(leap_parameters):
+            """Extend the loaded_water_files list given the leap_parameters"""
+            loaded_water_files.extend([water for water
+                                       in leap_parameters
+                                       if water in known_solvent_files])
```

Comment:
If you decide to keep `_OPENMM_LEAP_SOLVENT_FILES_MAP_R ` above, you could remove `known_solvent_files` and use directly `_OPENMM_LEAP_SOLVENT_FILES_MAP_R` to search by hash instead of going through the list.